modules/util/url.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
```

Comment:
A package named `util` is not a good idea maybe we could change this and begin from this PR?

Reply:
util does feel right for something like this but I assume the advice is to not re-use common stdlib names in general? What namespace would be preferred here? 

I'm not very good at naming things, perhaps just a generic 'gitea' for these utility type functions that are local to this project
```
gitea.PathEscapeSegments(myString)
``` 
Not very clever but at  least obvious to where it comes from. Lemme know what makes sense and I can update this PR. 

If the name needs more thought/feedback, I think it would be better to add this PR 'as is' and create a second PR for changing the name of the currently used modules/util since this will fix a few existing bugs.

Reply:
@lunny I think it's reasonable to continue using the util module here. We should refactor the module later as a separate pr.

Reply:
I think we could create a new package named `module/url` for all url related functions. But It's OK for another PR to do that.